gateway/handler_error.go

```diff
@@ -311,6 +314,29 @@ func (e *ErrorHandler) HandleError(w http.ResponseWriter, r *http.Request, errMs
 			log.WithError(err).Error("could not store analytic record")
 		}
 	}
+
+	// Print the transaction logs for error situations if enabled. Success transaction
+	// logs will be handled by the "handler_success.go"
+	if e.Spec.GlobalConfig.AccessLogs.Enabled {
+		// Don't print the full token, handle as obfuscated key or hashed key for security reasons
+		hashKeys := e.Gw.GetConfig().HashKeys
+		key := ""
+
+		if !hashKeys {
+			key = e.Gw.obfuscateKey(token)
+		} else {
+			key = storage.HashKey(token, hashKeys)
+		}
```

Comment:
This block just seems like utter spaghetti code. @sredxny why wouldn't obfuscateKey handle this?

https://github.com/TykTechnologies/tyk/blob/fd1a199bbf3b99796e13a0444db3bd5d0ec29e8f/gateway/log_helpers.go#L14